### ISSUE TITLE
`dart compile js` - help and accepted options inconsistent

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#48709]
dart cli prints a help message stating that the output selection is via `--output`, but this is not accepted.
The command accepts only `--out=<file name>`

Q. Do we fix cli or dart2js?
We could make dart2js accept both `out` and `output`.

```
$ out/ReleaseX64/dart-sdk/bin/dart compile js  --output=o1.js benchmarks/ObjectHash/dart/ObjectHash.dart
Compile Dart to JavaScript.

Usage: dart compile js [arguments] <dart entry point>
  -h, --help      Print this usage information (add -v for information about all options).
  -o, --output    Write the output to <file name>.
  -O<0,1,2,3,4>   Set the compiler optimization level (defaults to -O1).
  

Error: Unknown option '--output=o1.js'.
```

```
$ out/ReleaseX64/dart-sdk/bin/dart compile js --out=o1.js benchmarks/ObjectHash/dart/ObjectHash.dart 
Info: Compiling with sound null safety
Compiled 9,569,357 characters Dart to 185,845 characters JavaScript in 1.24 seconds
Dart file benchmarks/ObjectHash/dart/ObjectHash.dart compiled to JavaScript: o1.js
```

**Comment by sigmundch** [comment]
@rakudrama - we changed the dart cli to delegate everything to the dart2js snapshot, including help messages and flag parsing. So this help message is coming from dart2js itself. We changed it to unify with the style of other tools (like `dart compile exe -h`)

We could fix the help message [here](https://github.com/dart-lang/sdk/blob/5234076fbed26efc1e52a0621c27888082792533/pkg/compiler/lib/src/dart2js.dart#L1148) to say `out` or adjust the flag parsing. I like the idea of accepting `--output` to start bringing us closer to what the other CLI tools accept.
